### PR TITLE
Added the Giant Squid

### PR DESCRIPTION
**Pull request by Mailaender**
without `SquidGrab` only `SquidPunch`. See http://cnc.wikia.com/wiki/Giant_squid and https://gist.github.com/Mailaender/c2c5ff37a8927d0bc452 for reference. Also fixes https://github.com/OpenRA/ra2/issues/132.

### CODE REVIEW

File: rules/soviet-naval.yaml [hunk 1]
```diff
@@ -132,3 +132,81 @@ hyd:
...
+	DeathSounds:
+	MustBeDestroyed:
+		RequiredForShortGame: true
+	QuantizeFacingsFromSequence:
+		Sequence: stand
+	Chronoshiftable:
+		ExplodeInstead: true
+	RenderDetectionCircle:
+	RenderSprites:
+	WithInfantryBody:
```
Comment (GraionDilach) [inline rules/soviet-naval.yaml]: This should IMO use WithFacingSpriteBody instead (Squid animated played the move anim on a slower rate while standing still and IIRC WFSB can pull that off).

Comment (Mailaender) [inline rules/soviet-naval.yaml]: How do I setup `AttackSequence` using `WithFacingSpriteBody`?

Comment (GraionDilach) [inline rules/soviet-naval.yaml]: Use `WithAttackAnimation`.

Comment (Mailaender) [inline rules/soviet-naval.yaml]: Tried it. `WithFacingSpriteBody` won't play an animation during movement.

File: rules/soviet-naval.yaml [hunk 2]
```diff
@@ -132,3 +132,81 @@ hyd:
...
+		Range: 5c0
+	HiddenUnderFog:
+	ActorLostNotification:
+		Notification: NavalUnitLost
+	Guard:
+		Voice: Attack
+	Guardable:
+	DeathSounds:
+	MustBeDestroyed:
+		RequiredForShortGame: true
```
Comment (abcdefg30) [inline rules/soviet-naval.yaml]: This doesn't look correct.

File: rules/soviet-naval.yaml [hunk 3]
```diff
@@ -132,3 +132,81 @@ hyd:
...
+	QuantizeFacingsFromSequence:
+		Sequence: stand
+	Chronoshiftable:
+		ExplodeInstead: true
+	RenderDetectionCircle:
+	RenderSprites:
```
Comment (abcdefg30) [inline rules/soviet-naval.yaml]: This should be already inherited from `^SpriteActor`.

File: rules/soviet-naval.yaml [hunk 4]
```diff
@@ -132,3 +132,81 @@ hyd:
...
+	WithInfantryBody:
+		StandSequences: stand
+		AttackSequence: punch
+	WithDeathAnimation:
+		UseDeathTypeSuffix: false
```
Comment (abcdefg30) [inline rules/soviet-naval.yaml]: I somehow think we can drop that. The squids didn't have death animations and the weapon explosion should hide that anyway.

Comment (GraionDilach) [inline rules/soviet-naval.yaml]: What explosion? Squids and Dolphins didn1t explode at all.

Comment (abcdefg30) [inline rules/soviet-naval.yaml]: Weapon (impact) causing its death.

Comment (GraionDilach) [inline rules/soviet-naval.yaml]: The animals have a death animation though. This is completely legit.

Comment (abcdefg30) [inline rules/soviet-naval.yaml]: I don't remember one. Anyway, do you know where we can find the squids death animation then?

Comment (MustaphaTR) [inline rules/soviet-naval.yaml]: Squids don't have a death anim. Dolphins have a unused jumping anim but that is possibly cheer, not death.

Comment (GraionDilach) [inline rules/soviet-naval.yaml]: Hm, now I rechecked. Apparently I remembered wrong and they do lack death anims - the YR dinos got such which probably plays a role why I remembered these also got such I guess.

Comment (MustaphaTR) [inline rules/soviet-naval.yaml]: Dinos are infantry types. Squird amd dolphins are vehicle.

**Comment by Mailaender** [comment]
Updated.

File: rules/soviet-naval.yaml [hunk 5]
```diff
@@ -132,3 +132,76 @@ hyd:
...
+		TargetTypes: Underwater
+		UpgradeTypes: underwater
+		UpgradeMinEnabledLevel: 1
+		IsPlayerPalette: true
+	DetectCloaked:
+		CloakTypes: Underwater
+		Range: 5c0
+	HiddenUnderFog:
+	ActorLostNotification:
+		Notification: NavalUnitLost
```
Comment (abcdefg30) [inline rules/soviet-naval.yaml]: This isn't correct anymore since #77 was merged.

**Comment by abcdefg30** [comment]
Some other things I noticed:
- The self heal is a bit too fast
- It doesn't do damaged to dolphins/subs (but dolphins don't do damage to it, too)
- Weren't both squids and dolphins repairable (at the naval yards)?

Looks good to me otherwise.

**Comment by MustaphaTR** [comment]
> Weren't both squids and dolphins repairable (at the naval yards)?

Yes, they were.

**Comment by Mailaender** [comment]
Updated. Also fixed the weapon sounds. I always miss the .wav file extension.

**Comment by abcdefg30** [comment]
Somehow squids still don't do any damage.

You can also still damage land units with dolphins, but I don't have in mind if this is fixable yet:
dolphinlandattack

**Comment by MustaphaTR** [comment]
> You can also still damage land units with dolphins, but I don't have in mind if this is fixable yet:

This happens in  orginal RA2 too if you target them from a short piece of land that they can reach to other side.

File: rules/soviet-naval.yaml (hunk 4 above)
Comment (Mailaender) [inline rules/soviet-naval.yaml]: I currently use the jumping as idle which is a bit weird as it doesn't resurface. Cheer sounds like a better plan. That is used when the player won?

Comment (MustaphaTR) [inline rules/soviet-naval.yaml]: Cheer is used in missions by a action numbered as 113 (iirc). Also you can do cheer by clicking C or whatever you set it. There is also a unused but enablable by editing cheer Advaned Command Bar button for Cheer.

**Comment by Mailaender** [comment]
It is fixable, though not here, but at https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Effects/AreaBeam.cs

**Comment by GraionDilach** [comment]
Needs a rebase. Also please set `ValidTargets` on the Dolphin warheads as well (not just on the weapon), that will degrade the bug to purely visuals.

**Comment by GraionDilach** [comment]
Oh, `SquidPunch` also needs the `ValidTargets` defined on the warhead as well.

**Comment by Mailaender** [comment]
Rebased and added more `ValidTargets`.

File: weapons.yaml [hunk 6]
```diff
@@ -905,6 +907,20 @@ SonicZap:
...
 		DamageTypes: Prone50Percent, TriggerProne, ExplosionDeath
+		ValidTargets: Water, Underwater
+
+SquidPunch:
+	ReloadDelay: 32
+	Range: 1c856
+	Report: vsquat1a.wav
+	ValidTargets: Water, Underwater
+	Projectile: Bullet
+		Speed: 30
```
Comment (abcdefg30) [inline weapons.yaml]: This adds a weird delay in between the animation and the impact.
How about using a higher value (maybe `3000`, don't know if that's already overkill, though)?

**Comment by abcdefg30** [comment]
Looks good to me otherwise. :+1: / :white_check_mark: after above is fixed.

File: weapons.yaml (hunk 6 above)
Comment (Mailaender) [inline weapons.yaml]: I believe 100 is instant or at least normal bullet speed. Updated.

Comment (GraionDilach) [inline weapons.yaml]: 100c0 is instant, not 100.

**Comment by abcdefg30** [comment]
Thanks!

**Pull request merged**